scripts/python/packages/parthenon_tools/parthenon_tools/phdf.py

```diff
@@ -395,6 +432,106 @@ def Get(self, variable, flatten=True):
 
         return self.varData[variable][:]
 
+    def GetComponents(self, components, flatten=True):
+        """
+        Reads data for the named components from file.
+
+        Returns components as a dictionary of {component:component_data}
+
+        Throws an exception if the components do not exist in the file
+
+        Default is to return a flat array of length TotalCells.  However if
+        flatten is set to False, a 4D for each component array is returned that
+        has dimensions [NumBlocks, Nz, Ny, Nx] where NumBlocks is the total
+        number of blocks, and Nz, Ny, and Nx are the number of cells in the z,
+        y, and x directions respectively.
+        """
```

Comment:
I'm a little confused by the name "component" here. Can we add in a docstring somewhere what this word means precisely?

It looks to me like it means either a scalar dataset OR it means, e.g., `vector_x`?

Reply:
These are components within a dataset (or spread across multiple datasets). For AthenaPK, these are `Density`, `MomentumDensity1`,`TotalEnergyDensity`, etc.  They get written to `Info/ComponentNames`. There's documentation on how these get added: https://github.com/lanl/parthenon/blob/develop/docs/outputs.md#preparing-outputs-for-yt I'll add to the docstring.

Reply:
So do we need to add the component names for yt for this machinery to work?

Reply:
No, if you don't provide component names, Parthenon will give default component names that match the OutputDatasetNames. Calling `GetComponents` with dataset names in that case will return the dataset in its scalar/vector shape. `compare_analytic` will also happily accept those dataset names and managing components of the datasets is up to the user when providing the analytic functions.

I originally implemented the component names for `yt`, but later found them useful for `AthenaPK` tests.